src/features/instantiate/containers/InstanceSelection.tsx

```diff
@@ -43,15 +43,19 @@ function getDateDisplayString(timeStamp: string) {
 export function InstanceSelection({ instances, onNewInstance }: IInstanceSelectionProps) {
   const { data: applicationMetadata } = useApplicationMetadataQuery();
   const instanceSelectionOptions = applicationMetadata?.onEntry?.instanceSelection;
+  const selectedIndex = instanceSelectionOptions?.defaultSelectedOption;
   const { lang, langAsString, language } = useLanguage();
   const mobileView = useIsMobileOrTablet();
   const rowsPerPageOptions = instanceSelectionOptions?.rowsPerPageOptions ?? [10, 25, 50];
-  const defaultSelectedOption = instanceSelectionOptions?.defaultSelectedOption ?? 0;
 
+  const doesIndexExist = (selectedIndex: number) =>
+    rowsPerPageOptions.length - 1 >= selectedIndex && selectedIndex >= 0 ? selectedIndex : 0;
+
+  const defaultSelectedOption = selectedIndex ? doesIndexExist(selectedIndex) : 0;
```

Comment:
Maybe this is more correct?
```suggestion
  const defaultSelectedOption = doesIndexExist(selectedIndex) ? selectedIndex : 0;
```

Reply:
Yeah I kinda agree, however need to make sure that selectedIndex is not undefined before passing to rowsPerPageOptions.  Does this looks better?

```
  const doesIndexExist = (selectedIndex: number) =>
    rowsPerPageOptions.length - 1 >= selectedIndex && selectedIndex >= 0;

  const defaultSelectedOption = selectedIndex && doesIndexExist(selectedIndex) ? selectedIndex : 0;
  ...
  const [rowsPerPage, setRowsPerPage] = useState(rowsPerPageOptions[defaultSelectedOption]);
```

Reply:
I think it makes the most sense if `doesIndexExist` checks both that it is within the legal range and that it, well, exists (not undefined)

Reply:
Indeed, but for some reason when it checks that it exists inside in `doesIndexExist` I get an typescript error in `rowsPerPageOptions[defaultSelectedOption]` that undefined cannot be used as an index type. 

```
const doesIndexExist = (selectedIndex: number | undefined) =>
    selectedIndex && rowsPerPageOptions.length - 1 >= selectedIndex && selectedIndex >= 0;

  const defaultSelectedOption = doesIndexExist(selectedIndex) ? selectedIndex : 0;
  const [rowsPerPage, setRowsPerPage] = useState(rowsPerPageOptions[defaultSelectedOption]);

```

Reply:
ahh of course. you could get around that by making the function a typeguard i guess. just set the return type as `selectedIndex is number`.

Reply:
but whatever, sorry for getting hung up on trivial details 😅

Reply:
No problem! :D That's a helpful tip! 
This works:
```
  const doesIndexExist = (selectedIndex: number | undefined): selectedIndex is number =>
    selectedIndex !== undefined && rowsPerPageOptions.length - 1 >= selectedIndex && selectedIndex >= 0;

  const defaultSelectedOption = doesIndexExist(selectedIndex) ? selectedIndex : 0;
 ```